app/wasm/message_plugin.go

```diff
@@ -71,17 +71,22 @@ func (m *MintTokenMessenger) mintTokens(ctx sdk.Context, contractAddr sdk.AccAdd
 	if err != nil {
 		return nil, nil, sdkerrors.Wrap(err, "sending newly minted coins from message")
 	}
-
 	return nil, nil, nil
 }
 
-func (m *MintTokenMessenger) swapTokens(ctx sdk.Context, contractAddr sdk.AccAddress, swap *wasmbindings.SwapMsg) ([]sdk.Event, [][]byte, error) {
-	_, err := performSwap(m.gammKeeper, ctx, contractAddr, swap)
-	return nil, nil, sdkerrors.Wrap(err, "gamm estimate price exact amount out")
```

Comment:
This is shorthand for your code below... sdkerrors.Wrap(err, ...) return nil if err == nil.

But the refactor is cleaner, and allows adding events if we wish on success case.